### PR TITLE
tests(derive_code_mappings): Move more logic to the base class

### PR DESCRIPTION
**Pull request by armenzg**
And also use the generate_data function

### CODE REVIEW

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 1]
```diff
@@ -13,7 +13,6 @@
 from sentry.tasks.derive_code_mappings import derive_code_mappings, identify_stacktrace_paths
 from sentry.testutils import TestCase
 from sentry.testutils.helpers import with_feature
-from sentry.testutils.helpers.datetime import before_now, iso_format
```
Comment (armenzg) [inline tests/sentry/tasks/test_derive_code_mappings.py:16]: We were adding a lot of metadata for Python events but our code doesn't care about it.

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 2]
```diff
@@ -169,30 +170,13 @@ class TestPythonDeriveCodeMappings(BaseDeriveCodeMappings):
     def setUp(self):
         super().setUp()
         self.test_data = {
-            "message": "Kaboom!",
```
Comment (armenzg) [inline tests/sentry/tasks/test_derive_code_mappings.py:172]: Removing metadata not using in any of the code to simplify reading this.

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 3]
```diff
@@ -269,11 +248,10 @@ def test_derive_code_mappings_single_project(
         assert code_mapping.exists()
         assert code_mapping.first().automatically_generated is True
 
-    def test_skips_supported_projects(self):
-        new_data = deepcopy(self.test_data)
-        new_data["platform"] = "elixir"
-        event = self.store_event(data=new_data, project_id=self.project.id)
-        assert derive_code_mappings(self.project.id, event.data) is None
+    def test_skips_not_supported_platforms(self):
```
Comment (armenzg) [inline tests/sentry/tasks/test_derive_code_mappings.py:251]: Correct name of test.

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 4]
```diff
@@ -269,11 +248,10 @@ def test_derive_code_mappings_single_project(
...
+        data = self.generate_data([{}], platform="elixir")
```
Comment (armenzg) [inline tests/sentry/tasks/test_derive_code_mappings.py:252]: The base class has this function which simplifies saving an event for a test.

Comment (snigdhas) [reply tests/sentry/tasks/test_derive_code_mappings.py:271]: TIL!

**Review by snigdhas — APPROVED** [review]

File: tests/sentry/tasks/test_derive_code_mappings.py (hunk 4 above)
Comment (armenzg) [reply tests/sentry/tasks/test_derive_code_mappings.py:271]: > TIL!

FYI This is not part of `TestCase` but the class I created in this module.

**Pull request merged**